tests/unit/utils/cache-test.js

```diff
@@ -0,0 +1,154 @@
+import { module, test } from 'qunit';
+import { QueryCache } from 'ember-scoped-query-cache/utils';
+
+module('Unit | Utility | query-queryCache');
+
+test('Creates queryCache object with internal map', function (assert) {
+  let queryCache = new QueryCache();
+
+  assert.ok(queryCache._internalCache);
+});
+
+test('QueryCache#add skips adding if key is undefined', function(assert) {
```

Comment:
i believe this should be an error.

if the arity of `add` is not exactly 3 throw.

```js
if (arguments.length !== 3) {
  throw TypeError
}
```

or as an assertion

```js
assert('something about invalid arguments to queryCache.add', arguments.length !== 3);
```